overlord/snapstate/handlers.go

```diff
@@ -1564,7 +1564,23 @@ func (m *SnapManager) doLinkSnap(t *state.Task, _ *tomb.Tomb) (err error) {
 
```

Comment:
do we reboot again along the undo path? do we need to do something about that? I suppose at that point we will have reboot more than once anyway

Reply:
Yes, there would be restart requests for the base and for the kernel.